lib/qmp.ml

```diff
@@ -150,6 +151,12 @@ let message_of_string str =
       json |> arguments |> U.member "enable" |> U.to_bool
     in
     let add_fd json =
+      match json |> arguments  with
+      | `Assoc _  ->  json |> arguments |> U.member "fdset-id" |> U.to_int_option
+      | `Null -> None
+      | _ -> failwith "Unknown type"
+    in
```

Comment:
If both 'arguments' and 'arguments.fdset-id' are optional, I think what you want to have here is:
```
 let arguments = U.member "arguments" in
+let flatten_option = function None -> None | Some x -> x in
...
-json |> arguments |> U.member "fdset-id" |> U.to_int
+json |> arguments |> U.to_option (fun x -> x |> U.member "fdset-id" |> U.to_int_option) |> flatten_option
```
@lindig what do you think?